x-pack/plugins/ml/server/models/data_frame_analytics/validation.ts

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { IScopedClusterClient } from 'kibana/server';
+import { getAnalysisType } from '../../../common/util/analytics_utils';
+import {
+  INCLUDED_FIELDS_THRESHOLD,
+  MINIMUM_NUM_FIELD_FOR_CHECK,
+  FRACTION_EMPTY_LIMIT,
+  TRAINING_DOCS_LOWER,
+  TRAINING_DOCS_UPPER,
+  VALIDATION_STATUS,
+} from '../../../common/constants/validation';
+import { getDependentVar } from '../../../common/util/analytics_utils';
+import { extractErrorMessage } from '../../../common/util/errors';
+import { SearchResponse7 } from '../../../common';
+import { DataFrameAnalyticsConfig } from '../../../common/types/data_frame_analytics';
+
+interface MissingAgg {
+  [key: string]: {
+    doc_count: number;
+  };
+}
+interface CardinalityAgg {
+  [key: string]: {
+    value: number;
+  };
+}
+
+type ValidationSearchResult = Omit<SearchResponse7, 'aggregations'> & {
+  aggregations: MissingAgg | CardinalityAgg;
+};
+
+const defaultQuery = { match_all: {} };
+
+const trainingPercentHeading = i18n.translate(
+  'xpack.ml.models.dfaValidation.messages.trainingPercentHeading',
+  {
+    defaultMessage: 'Training percent',
+  }
+);
+const analysisFieldsHeading = i18n.translate(
+  'xpack.ml.models.dfaValidation.messages.analysisFieldsHeading',
+  {
+    defaultMessage: 'Analysis fields',
+  }
+);
+const dependentVarHeading = i18n.translate(
+  'xpack.ml.models.dfaValidation.messages.dependentVarHeading',
+  {
+    defaultMessage: 'Dependent variable',
+  }
+);
+const dependentVarWarningMessage = {
+  id: 'dep_var_check',
+  text: '',
+  status: VALIDATION_STATUS.WARNING,
+  heading: dependentVarHeading,
+};
+const analysisFieldsWarningMessage = {
+  id: 'analysis_fields',
+  text: '',
+  status: VALIDATION_STATUS.WARNING,
+  heading: analysisFieldsHeading,
+};
+
+function getTrainingPercentMessage(trainingDocs: number) {
+  let trainingPercentMessage;
+
+  if (trainingDocs >= TRAINING_DOCS_UPPER) {
+    trainingPercentMessage = {
+      id: 'training_percent_high',
+      text: i18n.translate('xpack.ml.models.dfaValidation.messages.highTrainingPercentWarning', {
+        defaultMessage:
+          'High number of training docs may result in long running jobs. Try reducing the training percent.',
+      }),
+      status: VALIDATION_STATUS.WARNING,
+      heading: trainingPercentHeading,
+    };
+  } else if (trainingDocs <= TRAINING_DOCS_LOWER) {
+    trainingPercentMessage = {
+      id: 'training_percent_low',
+      text: i18n.translate('xpack.ml.models.dfaValidation.messages.lowTrainingPercentWarning', {
+        defaultMessage:
+          'Low number of training docs may result in inaccurate models. Try increasing the training percent or using a larger dataset.',
```

Comment:
```suggestion
          'A low number of training docs may result in inaccurate models. Try increasing the training percent or using a larger data set.',
```

Reply:
Added all message updates in 8458efe8cd2ff4ffe13319473cd357721dc40e89